### ISSUE TITLE
[ERROR] Running checkpoint against RuntimeBatchRequest with Sql query

### ISSUE THREAD

**Issue by guidoturtu** [fivetran/great_expectations#3593]
**Describe the bug**
Running a checkpoint against a RuntimeBatchRequest with an sql query configured. With the same batch and running a validator, there is no issue and execute correctly. However, when I try to run a checkpoint with that batch, and error of `batch_data` parameter that os trying to verify. Investigation, I realized that there is a parameter to check the `batch_data` parameter to make a deep copy of pandas data frame for making validations. But that makes that cannot run with sql query

**To Reproduce**
1. Create GE venv
2. Init Repo
3. Define Postgre Datasource
4. Create expectations suite against RuntimeBatchRequest
5. Create checkpoint
6. Run checkpoint
7. See error

**Expected behavior**
Run `checkpoint` with expectations suite defined with a `RuntimeBatchRequest` and `query` with errors.

**Environment (please complete the following information):**
 - Operating System: [e.g. Linux and MacOS ]
 - Great Expectations Version: [e.g. 0.13.37]

**Additional context**
Error:
`````python
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
/var/folders/8k/3f3_q_dj3sx7fkrcxz2whdjm0000gn/T/ipykernel_50786/2101110360.py in <module>
     24 
     25 # Create checkpoint
---> 26 checkpoint = Checkpoint(
     27     name = my_checkpoint_name,
     28     data_context = context,
​
~/F14/envs/greatexpectations/lib/python3.8/site-packages/great_expectations/checkpoint/checkpoint.py in __init__(self, name, data_context, config_version, template_name, module_name, class_name, run_name_template, expectation_suite_name, batch_request, action_list, evaluation_parameters, runtime_configuration, validations, profilers, validation_operator_name, batches, ge_cloud_id, expectation_suite_ge_cloud_id)
     82                 if "batch_request" in val:
     83                     if isinstance(val["batch_request"], RuntimeBatchRequest):
---> 84                         batch_data = val["batch_request"]["runtime_parameters"][
     85                             "batch_data"
     86                         ]
​
KeyError: 'batch_data'
`````

**Comment by talagluck** [comment]
Thanks for submitting this issue, @guidoturtu! This is on our roadmap to fix, and we will be in touch when it is resolved.

**Comment by NathanFarmer** [comment]
Hi @guidoturtu! This issue will be resolved by one of the following PRs:
- #3680 
- #3713 
- #3718 
- #3723 

The fix will be included as part of GE release 0.13.44 later this week.
